source/magpie/index.d

```diff
@@ -5,79 +5,77 @@ Structure for DataFrame Indexing
 +/
 struct Index
 {
+private:
+    struct Indexing
+    {
+        string[] titles;
+        string[][] index;
+        int[][] codes;
+    }
+
+public:
     /// To know if data is multi-indexed
     bool isMultiIndexed = false;
 
-    /// Stores title to refer to each index level
-    string[] rtitles = [];
-    /// The indexes themselves
-    string[][] indexes = [];
-    /// Codes to map the above index to their positions
-    int[][] rcodes = [];
-
-    /// Titles for each column level
-    string[] ctitles = [];
-    /// The column indexes themself
-    string[][] columns = [];
-    /// Codes to map the index of above column to their position
-    int[][] ccodes = [];
+    /// Row and Column indexing
+    Indexing[2] indexing;
```

Comment:
add 
```
    ref row() { return indexing[0]; }
    ref column() { return indexing[1]; }
```
and use those in place of `indexing[0]` and `indexing[1]`

Reply:
Addressed in 2b3a9b1

Reply:
Thanks. Much easier to read.